src/main.cpp

```diff
@@ -0,0 +1,74 @@
+#include <QCoreApplication>
+#include <QDebug>
+#include <QStack>
+
+#include <QListIterator>
+#include <algorithm>
+
+int main(int argc, char *argv[])
+{
+    QCoreApplication a(argc, argv);
+
+    QList<QString> names;
+    names << "Mary" << "Ann" << "Charlynn" << "Marylynn" << "Margaret"
+          << "Kate" << "Rose" << "Gwendolyn";
+
+
+    // TODO: using Java-style iterators print the longest name from the list
+    // use QString::arg to output message with this name
+    QListIterator<QString> it(names);
+    QString longestName = it.next();
+    while (it.hasNext()) {
+        QString tmp = it.next();
```

Comment:
Please do not use "temp"/"tmp" as a name for local variables.
Every variable with automatic storage duration is temporary.